---
 prog/detect/sensors-detect |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/prog/detect/sensors-detect
===================================================================
--- a/prog/detect/sensors-detect	2011-12-07 18:49:04.895761902 -0500
+++ b/prog/detect/sensors-detect	2011-12-07 18:39:26.592168000 -0500
@@ -6513,7 +6513,7 @@
 
 		print "Successful!\n\n";
 		print "Monitoring programs won't work until the needed modules are\n".
-		      "loaded. You may want to run '/etc/init.d/module-init-tools start'\n".
+		      "loaded. You may want to run 'service module-init-tools start'\n".
 		      "to load them.\n";
 	}
 	print "\n";
